### PR TITLE
Auth: Add prometheus metrics for case insensitive ids

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
We want to have prometheus metrics for authinfo to collection duplicate users this PR:

- Adds prometheus metrics for duplicate user entries stats from authinfo
- Changes stats collection time from 30 min to 4 hours as we want an indication and do not want to smash the db
- Adds stats and prometheus metrics for `mixed_cased_users`

next
- [x] add stats and prometheus metrics for `mixed_cased_users `

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana/issues/50567

Fixes #

**Special notes for your reviewer**:

- Is there a way to test metrics stats being populated without waiting for the timer 🤔 ?

### CODE REVIEW

**Review by Jguer — CHANGES_REQUESTED** [review]
Missing a Run method to satisfy the background service interface like in (https://github.com/grafana/grafana/pull/51831/files#diff-d58f445e4c726d7d253fc1c06950361235a3a1a297b18edb36f0bffbd653751fR53) 

and the registration of the background service so it actually runs every 4h https://github.com/grafana/grafana/pull/51831/files#diff-cee3378f62a27dd53c0d3be445f4f64a482a8be79fa83938b87cfda290b8fd96L70

Usage statistics do not appear after these changes, registration seems to be missing

File: pkg/services/login/authinfoservice/service.go [hunk 1]
```diff
@@ -25,7 +25,13 @@ func ProvideAuthInfoService(userProtectionService login.UserProtectionService, a
 		authInfoStore:         authInfoStore,
 		logger:                log.New("login.authinfo"),
 	}
-	usageStats.RegisterMetricsFunc(authInfoStore.CollectLoginStats)
+	// collect login stats
+	go func() {
```
Comment (Jguer) [inline pkg/services/login/authinfoservice/service.go:29]: This causes usage statistics to not have the stats (checked using `/api/admin/usage-report-preview`).

`usageStats.RegisterMetricsFunc(authInfoStore.CollectLoginStats)` should still be called here and a background service registered for the prometheus stats

Comment (eleijonmarck) [reply pkg/services/login/authinfoservice/service.go]: > usageStats.RegisterMetricsFunc(authInfoStore.CollectLoginStats) should still be called here and a background service registered for the prometheus stats

I see, thanks for the thorough review to check the api/admin/usage-report endpoint.

Comment (eleijonmarck) [reply pkg/services/login/authinfoservice/service.go]: took back registerMetricsFunc

File: pkg/services/login/authinfoservice/database/stats.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	`
+	return sqlQuery
+}
+
+func (s *AuthInfoStore) mixedCasedUsers(ctx context.Context) string {
+	userDialect := db.DB.GetDialect(s.sqlStore).Quote("user")
+	// this query counts how many users have upper case and lower case login or emails.
+	// why
+	// users login via IDP or service providers get upper cased domains at times :shrug:
+	sqlQuery := `SELECT COUNT(*) FROM ` + userDialect + ` WHERE (LOWER(login) != login OR lower(email) != email)`
```
[the remark this answers is not in the document]
Comment (eleijonmarck) [reply pkg/services/login/authinfoservice/database/stats.go]: since cherry-pick I implemented this in commit: 9842b1926f , instead of suggestion

**Review by Jguer — APPROVED** [review]
I've tested this commit locally:

- [x] usage stats work
- [x] prometheus metrics work and are not updated on successive calls

Thanks for all of the effort into implementing this when it had such precise parameters and constraints

**Pull request merged**